### DISCUSSION TITLE
Support for MongoDB as an additional database in openFGA

### DISCUSSION THREAD

**Discussion by S4nfs** [openfga/community#373] [Ideas]
### Checklist

- [X] I agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md).

### Describe the problem you'd like to have solved

I've identified a crucial feature that's currently missing & might increase productivity, which became apparent due to my production app utilizing MongoDB Atlas as its database. Currently, only MySQL and PostgreSQL are supported. As a result, I had to provision a separate cloud server specifically for MySQL, which is my preferred current choice for openFGA. This not only increases the complexity of managing the database server but also consumes additional time optimizing both the databases. 

### Describe the ideal solution

I would like to suggest the addition of MongoDB as an add-on to OpenFGA. Ideally, this would include straightforward data migration options via the OpenFGA CLI. Thanks.

**Comment by miparnisari** [comment]
Hi @S4nfs, we are discussing internally within the team how to make the addition of new supported databases a smoother and faster experience, both for the community and the maintainers. Sadly, we don't have a specific date for when the design will be finalized, but I'll post here when we have updates. Stay tuned :)

**Comment by danielloader** [comment]
I imagine it'll be possible, and the right time to revisit this if/when https://github.com/openfga/roadmap/issues/49 makes some traction. In principle if you can support DynamoDB, MongoDB isn't a huge leap from there.

**Comment by JMRMourao** [comment]
Hello team,

This is exactly what i was looking for, since you are using JSON documents it would be relevant to have MongoDB as a specific connector as it would indeed fit as a great addition.

**Comment by danielloader** [comment]
The issue isn't storing the json/tuples per se, it's more the query patterns are quite SQL friendly and not very NoSQL friendly at a guess.
